lib/ex_unit/lib/ex_unit/manifest.ex

```diff
@@ -4,19 +4,41 @@ defmodule ExUnit.Manifest do
   import Record
 
   defrecord :entry, [:last_run_status, :file]
-  @opaque t :: [{test_id, entry}]
+  defstruct entries: [], path: nil
```

Comment:
I know this somewhat of a nitpick but is the path really a property of the manifest?

A manifest, once loaded, can be written anywhere in disk, and not simply on the path it was originally created at. So it feels we are trying to couple two things that don't really belong together. :) And as a consequence the code got much more complex too. Simple functions like merge now need to extract values from structs, discard paths and what not.

Reply:
Although I understand now that you did it to avoid loading the manifest multiple times. I am thinking that it is ok to read the manifest twice when the `--only-failures` flag is given, since we have really optimized that.

So maybe a better way to go about this is to have a new function in `ExUnit.Filters` that returns the failed files from the manifest?

Another option is to have a function in `ExUnit.Filters` that reads the manifest and returns all failed files and all failed tests per module. This way, we won't tell ExUnit to run only last_status_run:failed, which we said we don't want to expose to developers, but instead we simply tell it which tests to run in any given module. With this approach, we don't need to touch the manifest in the runner at all.